tests/modules/lint/test_module_changes.py

```diff
@@ -0,0 +1,75 @@
+import pytest
+
+import nf_core.modules.lint
+
+from ...test_modules import TestModules
+
+
+class TestModuleChanges(TestModules):
+    """Test module_changes.py functionality"""
+
+    def test_module_changes_unchanged(self):
+        """Test module changes when module is unchanged"""
+        # Install a module that should be unchanged from the repository
+        assert self.mods_install.install("samtools/sort")
+
+        # Run lint on the unchanged module
+        module_lint = nf_core.modules.lint.ModuleLint(directory=self.pipeline_dir)
+        module_lint.lint(print_results=False, module="samtools/sort", key=["module_changes"])
+
+        # Check that module_changes test passed (no changes detected)
+        assert len(module_lint.failed) == 0, f"Linting failed with {[x.__dict__ for x in module_lint.failed]}"
+
+        # Should have passed entries for files being up to date
+        passed_test_names = [test.lint_test for test in module_lint.passed]
+        assert "check_local_copy" in passed_test_names
+
+    def test_module_changes_modified_main_nf(self):
+        """Test module changes when main.nf is modified"""
+        # Install a module
+        assert self.mods_install.install("samtools/sort")
+
+        # Modify the main.nf file
+        main_nf_path = self.pipeline_dir / "modules" / "nf-core" / "samtools" / "sort" / "main.nf"
+        with open(main_nf_path, "a") as fh:
+            fh.write("\n// This is a test modification\n")
+
+        # Run lint on the modified module
+        module_lint = nf_core.modules.lint.ModuleLint(directory=self.pipeline_dir)
+        module_lint.lint(print_results=False, module="samtools/sort", key=["module_changes"])
+
+        # Check that module_changes test failed (changes detected)
+        assert len(module_lint.failed) > 0, "Expected linting to fail due to modified file"
+
+        # Should have failed entry for local copy not matching remote
+        failed_test_names = [test.lint_test for test in module_lint.failed]
+        assert "check_local_copy" in failed_test_names
+
+    def test_module_changes_modified_meta_yml(self):
+        """Test module changes when meta.yml is modified"""
+        # Install a module
+        assert self.mods_install.install("samtools/sort")
```

Comment:
looks like we always do this as the first step. best if we have it as a setup step.